src/shared_realm.hpp

```diff
@@ -31,6 +31,8 @@
 #include <memory>
 #include <thread>
 
+#include <realm/group_shared.hpp>
```

Comment:
If this is only needed because of the `_impl::RealmFriend` function bodies being inlined, it'd be preferable to move them, and this include, to the .cpp file instead.